oracle/src/services/gasPrice.js

```diff
@@ -73,17 +73,20 @@ async function start(chainId, fetchOnce) {
     throw new Error(`Unrecognized chainId '${chainId}'`)
   }
 
-  const fetchFn = gasPriceSupplierUrl === 'gas-price-oracle' ? null : () => fetch(gasPriceSupplierUrl)
+  let fetchFn = null
+  if (gasPriceSupplierUrl !== 'gas-price-oracle') {
+    fetchFn = () => fetch(gasPriceSupplierUrl, { timeout: 2000 })
```

Comment:
May you share more info why this timeout introduced?

Reply:
As far as I know, the fetch function does not have a default timeout for the HTTP request. Also, I believe that it is not a good idea to allow possibly infinite requests that won't be terminated after some time.

Reply:
Ah, OK. By some reason thought that it could be a timeout for a re-occurring process.